### PR TITLE
[NUI] Add Component Application

### PR DESCRIPTION
**Pull request by huiyueun**
### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI/src/public/NUIComponentApplication.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    /// <summary>
+    /// The class for supporting multi-components application model.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class NUIComponentApplication : CoreApplication
+    {
+        private Dictionary<Type, ComponentStateManger> _componentFactories = new Dictionary<Type, ComponentStateManger>();
+
+        /// <summary>
+        /// Initializes the ComponentApplicationBase class.
```
Comment (elishateng) [inline src/Tizen.NUI/src/public/NUIComponentApplication.cs]: Could I ask if ComponentApplicationBase means NUIComponentApplication ?

File: src/Tizen.NUI/src/internal/NUIWindowInfo.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
+using System;
+using Tizen.NUI;
+using Tizen.Applications.ComponentBased.Common;
+
+namespace Tizen.NUI
+{
+    /// <summary>
+    /// Window information class for ComponentBasedApplication
```
Comment (elishateng) [inline src/Tizen.NUI/src/internal/NUIWindowInfo.cs]: Could I ask if ComponentBasedApplicationmeans NUIComponentApplication ?

File: src/Tizen.NUI/src/public/NUIComponentApplication.cs (hunk 1 above)
Comment (huiyueun) [reply src/Tizen.NUI/src/public/NUIComponentApplication.cs]: fixed.

File: src/Tizen.NUI/src/internal/NUIWindowInfo.cs (hunk 2 above)
Comment (huiyueun) [reply src/Tizen.NUI/src/internal/NUIWindowInfo.cs]: fixed.

**Comment by dongsug-song** [comment]
no more review done, so merged.

**Pull request merged**